libnavigation-core/src/main/java/com/mapbox/navigation/core/accounts/BillingController.kt

```diff
@@ -382,31 +385,32 @@ internal class BillingController(
      * ignoring origin.
      */
     private fun getRemainingWaypointsOnRoute(routeProgress: RouteProgress?): List<Point>? {
-        return routeProgress?.navigationRoute?.let { route ->
-            val waypoints = route.waypoints
-            waypoints?.drop(
-                (waypoints.size - routeProgress.remainingWaypoints).coerceAtLeast(1)
-            )?.map {
-                it.location()
-            }
+        if (routeProgress == null) return null
+        return getWaypointsOnRoute(routeProgress.navigationRoute) {
+            drop((size - routeProgress.remainingWaypoints).coerceAtLeast(1))
         }
     }
 
     /**
      * Returns a list of [Point]s that mark ends of legs on the route,
      * ignoring origin.
      */
-    private fun getWaypointsOnRoute(navigationRoute: NavigationRoute): List<Point>? {
-        return navigationRoute.waypoints?.drop(1)?.map {
-            it.location()
-        }
+    private fun getWaypointsOnRoute(
+        navigationRoute: NavigationRoute,
+        dropper: List<Waypoint>.() -> List<Waypoint>
+    ): List<Point> {
+        return navigationRoute
+            .internalWaypoints()
+            .dropper()
+            .filterNot { it.isEVChargingWaypoint() }
+            .map { it.location }
```

Comment:
This `dropper` extension is a little bit broad. Could we narrow down the purpose?

```suggestion
    private fun getNonEVWaypointsOnRoute(
        navigationRoute: NavigationRoute,
        drop: Int = 0
    ): List<Point> {
        return navigationRoute
            .internalWaypoints()
            .drop(drop)
            .filterNot { it.isEVChargingWaypoint() }
            .map { it.location }
```

Reply:
We can't because in some cases we use list size to determine the number of items to be dropped. And I don't want to get waypoints twice: what if at some moment we change getting in one place and forget about the other?
Rewrote this bit.